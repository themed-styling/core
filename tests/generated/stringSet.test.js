import 'jest-styled-components'

import { stringSet } from '../../src/index.js'

import { testAllOn } from '../utilities.js'

testAllOn(stringSet, 'stringSet')
