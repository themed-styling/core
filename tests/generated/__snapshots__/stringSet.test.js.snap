// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringSet renders 0 1`] = `
Array [
  .c0 {
  string-set: 0;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: 0;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: 0;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: 0;
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders function 1`] = `
Array [
  .c0 {
  string-set: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: has;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: has;
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders invalid theme path 1`] = `
Array [
  .c0 {
  string-set: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: invalid.path.1;
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders negative number 1`] = `
Array [
  .c0 {
  string-set: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: -42365;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: -42365;
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders object [breakpoints + standard] 1`] = `
Array [
  .c0 {
  string-set: July;
}

@media screen and (min-width:November) {
  .c0 {
    string-set: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: September;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: November;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: August;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  .c0 {
  string-set: July;
}

@media screen and (min-width:November) {
  .c0 {
    string-set: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: September;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: November;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: August;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  .c0 {
  string-set: January;
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    string-set: December;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: April;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: May;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: June;
  }
}

<div
    className="c0"
  />,
  .c0 {
  string-set: January;
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    string-set: December;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: April;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: May;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders object [breakpoints] 1`] = `
Array [
  @media screen and (min-width:November) {
  .c0 {
    string-set: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {
  .c0 {
    string-set: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    string-set: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: November;
  }
}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    string-set: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    string-set: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    string-set: November;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders object [mediaQueries + standart] 1`] = `
Array [
  .c0 {
  string-set: July;
}

@media May {
  .c0 {
    string-set: August;
  }
}

@media April {
  .c0 {
    string-set: September;
  }
}

@media January {
  .c0 {
    string-set: November;
  }
}

@media April {
  .c0 {
    string-set: August;
  }
}

@media June {
  .c0 {
    string-set: November;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  .c0 {
  string-set: July;
}

@media May {
  .c0 {
    string-set: August;
  }
}

@media April {
  .c0 {
    string-set: September;
  }
}

@media January {
  .c0 {
    string-set: November;
  }
}

@media April {
  .c0 {
    string-set: August;
  }
}

@media June {
  .c0 {
    string-set: November;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  .c0 {
  string-set: April;
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    string-set: November;
  }
}

@media April {
  .c0 {
    string-set: January;
  }
}

@media January {
  .c0 {
    string-set: December;
  }
}

@media April {
  .c0 {
    string-set: April;
  }
}

@media June {
  .c0 {
    string-set: May;
  }
}

<div
    className="c0"
  />,
  .c0 {
  string-set: April;
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    string-set: November;
  }
}

@media April {
  .c0 {
    string-set: January;
  }
}

@media January {
  .c0 {
    string-set: December;
  }
}

@media April {
  .c0 {
    string-set: April;
  }
}

@media June {
  .c0 {
    string-set: May;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders object [mediaQueries] 1`] = `
Array [
  @media May {
  .c0 {
    string-set: July;
  }
}

@media April {
  .c0 {
    string-set: August;
  }
}

@media January {
  .c0 {
    string-set: September;
  }
}

@media April {
  .c0 {
    string-set: November;
  }
}

@media June {
  .c0 {
    string-set: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {
  .c0 {
    string-set: July;
  }
}

@media April {
  .c0 {
    string-set: August;
  }
}

@media January {
  .c0 {
    string-set: September;
  }
}

@media April {
  .c0 {
    string-set: November;
  }
}

@media June {
  .c0 {
    string-set: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    string-set: January;
  }
}

@media April {
  .c0 {
    string-set: December;
  }
}

@media January {
  .c0 {
    string-set: April;
  }
}

@media April {
  .c0 {
    string-set: May;
  }
}

@media June {
  .c0 {
    string-set: June;
  }
}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    string-set: January;
  }
}

@media April {
  .c0 {
    string-set: December;
  }
}

@media January {
  .c0 {
    string-set: April;
  }
}

@media April {
  .c0 {
    string-set: May;
  }
}

@media June {
  .c0 {
    string-set: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders postitive number 1`] = `
Array [
  .c0 {
  string-set: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: 42365;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: 42365;
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders string 1`] = `
Array [
  .c0 {
  string-set: July;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: July;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: June;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: June;
}

<div
    className="c0"
  />,
]
`;

exports[`stringSet renders undefined 1`] = `
Array [
  <div
    className=""
  />,
  <div
    className=""
  />,
  <div
    className=""
  />,
  <div
    className=""
  />,
]
`;

exports[`stringSet renders valid theme path 1`] = `
Array [
  .c0 {
  string-set: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  string-set: DarkSlateBlue;
}

<div
    className="c0"
  />,
]
`;
